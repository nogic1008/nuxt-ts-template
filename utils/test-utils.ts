/** Generate [0-9a-z]{length} string. */
export const generateRandomString = (length: number) =>
  [...Array(Number.isInteger(length) ? length : 12)]
    .map(() => (~~(Math.random() * 36)).toString(36))
    .join('')
