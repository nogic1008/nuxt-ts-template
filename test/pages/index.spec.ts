import { shallowMount, createLocalVue, Wrapper, mount } from '@vue/test-utils'
import Buefy from 'buefy'
import Index from '~/pages/index.vue'

const localVue = createLocalVue()
localVue.use(Buefy)

describe('pages/index.vue', () => {
  let wrapper: Wrapper<Index>

  beforeEach(() => {
    wrapper = shallowMount(Index, { localVue })
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(Index, { localVue })
    expect(wrapper.element).toMatchSnapshot()
  })
})
