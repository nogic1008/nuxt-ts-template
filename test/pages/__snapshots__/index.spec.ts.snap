// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/index.vue renders correctly 1`] = `
<section
  class="section"
>
  <div
    class="columns is-mobile"
  >
    <div
      class="column"
    >
      <div
        class="card"
      >
        <header
          class="card-header"
        >
          <p
            class="card-header-title has-text-grey"
          >
            
        Free
      
          </p>
        </header>
         
        <div
          class="card-content"
        >
          <div
            class="content has-text-centered"
          >
            <span
              class="icon has-text-primary is-large"
            >
              <i
                class="mdi mdi-github-circle mdi-48px"
              />
            </span>
          </div>
        </div>
         
        <footer
          class="card-footer"
        >
          <div
            class="card-footer-item"
          >
            <span>
              
      Open source on
      
              <a
                href="https://github.com/buefy/buefy"
              >
                
        GitHub
      
              </a>
            </span>
          </div>
        </footer>
      </div>
    </div>
     
    <div
      class="column"
    >
      <div
        class="card"
      >
        <header
          class="card-header"
        >
          <p
            class="card-header-title has-text-grey"
          >
            
        Responsive
      
          </p>
        </header>
         
        <div
          class="card-content"
        >
          <div
            class="content has-text-centered"
          >
            <span
              class="icon has-text-primary is-large"
            >
              <i
                class="mdi mdi-cellphone-link mdi-48px"
              />
            </span>
          </div>
        </div>
         
        <footer
          class="card-footer"
        >
          <div
            class="card-footer-item"
          >
            <span>
              <b
                class="has-text-grey"
              >
                
        Every
      
              </b>
              
      component is responsive
    
            </span>
          </div>
        </footer>
      </div>
    </div>
     
    <div
      class="column"
    >
      <div
        class="card"
      >
        <header
          class="card-header"
        >
          <p
            class="card-header-title has-text-grey"
          >
            
        Modern
      
          </p>
        </header>
         
        <div
          class="card-content"
        >
          <div
            class="content has-text-centered"
          >
            <span
              class="icon has-text-primary is-large"
            >
              <i
                class="mdi mdi-alert-decagram mdi-48px"
              />
            </span>
          </div>
        </div>
         
        <footer
          class="card-footer"
        >
          <div
            class="card-footer-item"
          >
            <span>
              
      Built with
      
              <a
                href="https://vuejs.org/"
              >
                
        Vue.js
      
              </a>
              
      and
      
              <a
                href="http://bulma.io/"
              >
                
        Bulma
      
              </a>
            </span>
          </div>
        </footer>
      </div>
    </div>
     
    <div
      class="column"
    >
      <div
        class="card"
      >
        <header
          class="card-header"
        >
          <p
            class="card-header-title has-text-grey"
          >
            
        Lightweight
      
          </p>
        </header>
         
        <div
          class="card-content"
        >
          <div
            class="content has-text-centered"
          >
            <span
              class="icon has-text-primary is-large"
            >
              <i
                class="mdi mdi-arrange-bring-to-front mdi-48px"
              />
            </span>
          </div>
        </div>
         
        <footer
          class="card-footer"
        >
          <div
            class="card-footer-item"
          >
            <span>
              
      No other internal dependency
    
            </span>
          </div>
        </footer>
      </div>
    </div>
  </div>
</section>
`;
